File: Arrays/NumberAppearingMoreThanKTimes/NumberAppearingMoreThanKTimes.js
class Count{
    constructor(e,c){
        this.e = e;
        this.c = c;
    }
}

function method1(arr,k){
    let countArr = [];
    let i=0,j,n=arr.length;
    for(let no of arr){
        
        for(i=0;i<countArr.length;i++){
            if(countArr[i].e==no){
                countArr[i].c += 1;
                break;
            }
        }
        if(i==countArr.length)
            if(countArr.length<k-1){
                let o = new Count(no, 1);
                countArr.push(o);
                continue;
            }
        if(i>=k-1){
            for(i=0;i<countArr.length;i++){
                if(countArr[i].c==0){
                    countArr[i].e = no;
                    countArr[i].c = 1;
                    break;
                }
            }
            if(i==k-1){
                for(i=0;i<countArr.length;i++){
                    countArr[i].c -= 1;
                }
            }
        }
    }
    console.log('countArr is ', countArr);
    let res = [];
    for(let obj of countArr){
        let co = 0;
        for(let no of arr){
            if(no==obj.e)
                co++;
        }
        if(co>n/k)
            res.push(obj.e);
    }
    return res;
}

let arr = [1, 1, 2, 2, 3, 5, 4,
    2, 2, 3, 1, 1, 1]
let k = 4;
console.log(method1(arr, k));

File: Arrays/NumberAppearingMoreThanKTimes/NumberAppearingMoreThanKTimes.go
package main

type num struct {
	ele   int
	count int
}

func method1(arr []int, k int) {
	countArr := make([]num, 0)
	n := len(arr)
	i := 0
	j := 0
	for i = 0; i < n; i++ {
		for j = 0; j < len(countArr); j++ {
			if countArr[j].ele == arr[i] {
				countArr[j].count += 1
				break
			}
		}
		if j == len(countArr) {
			temp := num{
				ele:   arr[i],
				count: 1,
			}
			countArr = append(countArr, temp)
		}
		if j >= k-1 {
			for l := 0; l < len(countArr); l++ {
				if countArr[l].ele != arr[i] {
					countArr[l].count -= 1
				}
			}
		}
	}
}

func main() {
	arr := []int{1, 1, 2, 2, 3, 5, 4,
		2, 2, 3, 1, 1, 1}
	k := 4
	method1(arr, k)
}
